src/current/v24.3/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,395 @@
+`BEGIN;`                         |                              |
+`UPDATE t2 SET v=4014 WHERE k=4;`| `BEGIN AS OF SYSTEM TIME '-30s';`|
+lock k=4                         | `SELECT * FROM t2 WHERE k=4;`| `BEGIN;`
+                                 | not waiting                  | `SELECT * FROM t2 WHERE k=4 FOR UPDATE NOWAIT;`
+                                 | k=4,v=4                      | error: could not obtain lock
+`COMMIT;`                        |                              | `UPDATE t2 SET v=4034 WHERE k=4;`
+success, k=4,v=4014              |                              | not waiting
+                                 | `COMMIT;`                    | `COMMIT;`
+                                 | success                      | failure                                 
+                                 |                              | `SELECT * FROM t2 WHERE k=4;`
+                                 |                              | k=4, v=4014
+
+### SQL statements for Example 2
+
+To reproduce Example 2 in CockroachDB to show how to remediate a waiting read and write experiencing lock contention, execute the following SQL statements in the given order in the specified terminal.
+
+**Terminal 1**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 4';  -- to distinguish between transactions
+BEGIN;
+UPDATE t2 SET v=4014 WHERE k=4; -- lock k=4
+~~~
+
+**Terminal 2**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 5';  -- to distinguish between transactions
+BEGIN AS OF SYSTEM TIME '-30s';
+SELECT * FROM t2 WHERE k=4; -- historical read
+~~~
+
+Transaction 5 does a historical read and should output the following:
+
+~~~
+  k | v
+----+----
+  4 | 4
+(1 row)
+~~~
+
+**Terminal 3**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 6';  -- to distinguish between transactions
+BEGIN;
+SELECT * FROM t2 WHERE k=4 FOR UPDATE NOWAIT; -- fail fast write
+UPDATE t2 SET v=4034 WHERE k=4;
+~~~
+
+The `SELECT ... FOR UPDATE NOWAIT` returns:
+
+~~~
+ERROR: could not obtain lock on row (k)=(4) in t2@t2_pkey
+SQLSTATE: 55P03
+ERROR: current transaction is aborted, commands ignored until end of transaction block
+SQLSTATE: 25P02
+~~~
+
+**Terminal 1**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+~~~
+
+**Terminal 2**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+~~~
+
+**Terminal 3**
+
+`COMMIT` Transaction 6. Since the `SELECT` statement in Transaction 6 generated an error, `COMMIT` is equivalent to `ROLLBACK`, which aborts the transaction and discards the `UPDATE`. Afterward, verify that the `UPDATE` was discarded.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+COMMIT;
+SELECT * FROM t2 WHERE k=4;
+~~~
+
+The `SELECT` statement should output the following:
+
+~~~
+  k |  v
+----+-------
+  4 | 4014
+(1 row)
+~~~
```

Comment:
This page [needs a 'See also' section](https://github.com/cockroachdb/docs/blob/main/StyleGuide.md#see-also), easiest way to populate is with the links you were using in the page above

Reply:
added